apps/src/lib/client/utils.rs

```diff
@@ -970,22 +977,10 @@ fn init_genesis_validator_aux(
     genesis_validator
 }
 
-async fn download_file(url: impl AsRef<str>) -> Vec<u8> {
+async fn download_file(url: impl AsRef<str>) -> reqwest::Result<Bytes> {
```

Comment:
We could maybe combine this fn with apps/src/lib/wasm_loader/mod.rs:325 (download_wasm() function) since there is some overlap, especially if we were to start downloading WASM in the client rather than the ledger as well (#1048)